browser-test/src/header.test.ts

```diff
@@ -1,56 +1,59 @@
-import {test, expect} from '@playwright/test'
+import {test, expect} from './support/civiform_fixtures'
 import {
-  createTestContext,
   loginAsTestUser,
   validateScreenshot,
   validateAccessibility,
+  logout,
 } from './support'
 
-test.describe('Header', () => {
-  const ctx = createTestContext(/* clearDb= */ false)
-
-  test('Not logged in, guest mode enabled', async () => {
-    const {page} = ctx
-    await validateScreenshot(
-      page.getByRole('navigation'),
-      'not-logged-in-guest-mode-enabled',
-    )
-  })
-
-  // TODO(#4360): add a "Not logged in, guest mode disabled" test once we
-  // can get to the programs page without logging in, for an entity without
-  // guest mode.
-
-  test('Logged in', async () => {
-    const {page} = ctx
-    await loginAsTestUser(page)
-    await validateScreenshot(page.getByRole('navigation'), 'logged-in')
+test.describe('Header', {tag: ['@uses-fixtures']}, () => {
+  /**
+   * @todo (#4360) add a "Not logged in, guest mode disabled" test once we can get to the programs page without logging in, for an entity without guest mode.
+   */
+  test('Check screenshots and validate accessibility', async ({page}) => {
+    await test.step('Take a screenshot as a guest', async () => {
+      await logout(page)
+      await validateScreenshot(
+        page.getByRole('navigation'),
+        'not-logged-in-guest-mode-enabled',
+      )
+    })
+
+    await test.step('Take a screenshot as the test user', async () => {
+      await loginAsTestUser(page)
+      await validateScreenshot(page.getByRole('navigation'), 'logged-in')
+    })
+
+    await test.step('Passes accessibility test', async () => {
+      await validateAccessibility(page)
+    })
```

Comment:
Do more asserts in a single test. It's recommended by Playwright and it shaved 3.5s off this file 

![image](https://github.com/civiform/civiform/assets/195162/845faa01-1bc0-4d40-8b71-b36ce5572f15)